### PR TITLE
ElectricSQL sync

### PR DESCRIPTION
**Pull request by samwillis**

### CODE REVIEW

File: packages/pglite-sync/eslint.config.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  {
+    languageOptions: {
+      globals: {
+        ...globals.browser,
+        ...globals.node,
+      },
+    },
+    rules: {
+      ...rootConfig.rules,
+      '@typescript-eslint/no-explicit-any': 'off',
```
Comment (msfstef) [inline packages/pglite-sync/eslint.config.js:18]: nit: it seems you only use `any` for an unused parameter so you might as well remove this exception I think

File: packages/pglite-sync/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+    aborter: AbortController
+  }> = []
+
+  const namespaceObj = {
+    syncShapeToTable: async (options: SyncShapeToTableOptions) => {
+      const aborter = new AbortController()
+      if (options.signal) {
+        // we new to have our own aborter to be able to abort the stream
+        // but still accept the signal from the user
+        options.signal.addEventListener('abort', () => {
```
Comment (msfstef) [inline packages/pglite-sync/src/index.ts:33]:
```suggestion
        options.signal.addEventListener('abort', () => aborter.abort(), { once: true })
```
Can we/should we use a "once" listener here?

File: packages/pglite-sync/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+  return {
+    namespaceObj,
+    close,
+  }
+}
+
+export function electricSync(options: ElectricSyncOptions) {
+  return {
+    name: 'ElectricSQL Sync',
+    setup: async (pg: PGliteInterface, _emscriptenOpts: any) => {
```
Comment (msfstef) [inline packages/pglite-sync/src/index.ts:106]: nit: can remove emscriptenOpts and the "any" type

File: packages/pglite-sync/test/sync.test.ts [hunk 4]
```diff
@@ -0,0 +1,7 @@
+// import { describe, it, expect, beforeEach } from 'vitest'
```
Comment (msfstef) [inline packages/pglite-sync/test/sync.test.ts:1]: For tests we might want to mock out some responses

**Review by msfstef — APPROVED** [review]
Looks great, nice work!

File: packages/pglite-sync/src/index.ts [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+      const stream = new ShapeStream({
+        ...options,
+        signal: aborter.signal,
+      })
+      stream.subscribe(async (messages) => {
+        if (debug) {
+          console.log('sync messages received', messages)
+        }
+        for (const message of messages) {
+          await applyMessageToTable({
```
Comment (KyleAMathews) [inline packages/pglite-sync/src/index.ts:46]: can we apply these all in one transaction?

**Review by KyleAMathews — APPROVED** [review]
Looks all very neat!

File: packages/pglite-sync/src/index.ts (hunk 5 above)
Comment (samwillis) [reply packages/pglite-sync/src/index.ts:46]: Good call. In future we need to batch them by the prefix of the offset (LSN) and do that accross shapes. So a full LSN is commited at once.

**Pull request merged**